core/scripts/cre/engine/main.go

```diff
@@ -12,16 +12,19 @@ import (
 
 	"github.com/smartcontractkit/chainlink/v2/core/capabilities"
 	"github.com/smartcontractkit/chainlink/v2/core/logger"
+	"github.com/smartcontractkit/chainlink/v2/core/services/workflows"
 )
 
 func main() {
 	var wasmPath string
 	var configPath string
 	var debugMode bool
+	var noDAG bool
 
 	flag.StringVar(&wasmPath, "wasm", "", "Path to the WASM binary file")
 	flag.StringVar(&configPath, "config", "", "Path to the Config file")
 	flag.BoolVar(&debugMode, "debug", false, "Enable debug-level logging")
+	flag.BoolVar(&noDAG, "no-dag", false, "Enable V2 engine (No DAG) mode")
```

Comment:
We don't need an explicit flag. Let's use the same logic as here: https://github.com/smartcontractkit/chainlink/blob/develop/core/services/workflows/syncer/handler.go#L471